app/models/services/commitment_props.rb

```diff
@@ -0,0 +1,322 @@
+                description: 'Multi select option',
+                # defaultValue: [],
+                choices: [
+                  'Indigenous peoples',
+                  'Local communities',
+                  'For-profit organisations',
+                  'Non-profit organisations',
+                  'Individual landowners',
+                  'Collaborative governance',
+                  'Joint governance',
+                  'Sub-national ministry or agency',
+                  'Other'
+                ],
+                otherText: 'Other'
+              },
+              {
+                type: 'comment',
+                name: 'jointGovernanceType',
+                visibleIf: "{managerIds} contains 'Joint governance'",
+                title: 'Specify the type of joint governance',
+                hideNumber: true
+              },
+              {
+                type: 'text',
+                name: 'responsibleGroup',
+                title: 'Name of the group responsible for decisions'
+              }
+            ],
+            title: 'General Info',
+            description:
+              '‘Commitment’ refers to what you intend to do, for example establishing a new protected area, or working to improve the management of an existing area. ‘Area’ refers to the geographical area that is (or will be) under management as a result of this commitment.’'
+          },
+          {
+            name: 'locationPage',
+            elements: [
+              {
+                type: 'tagbox',
+                name: 'countryIds',
+                title: 'Select commitment location by country',
+                description: 'Multi select option',
+                # defaultValue: [],
+                choicesByUrl: {
+                  url: 'https://surveyjs.io/api/CountriesExample'
+                },
+                optionsCaption: '-Select a country-'
+              },
+              {
+                type: 'expression',
+                name: 'latlongHeading',
+                title: 'Please provide the latitude and longitude of the area'
+              },
+              {
+                type: 'text',
+                name: 'latitude',
+                title: 'Latitude',
+                titleLocation: 'left',
+                hideNumber: true
+              },
+              {
+                type: 'text',
+                name: 'longitude',
+                title: 'Longitude',
+                titleLocation: 'left',
+                hideNumber: true
+              },
+              {
+                type: 'file',
+                name: 'geospatialFilename',
+                title:
+                  'If you have spatial data you would like to share, please upload it here.',
+                description: '(Optional field)',
+                hideNumber: true,
+                allowImagesPreview: false,
+                maxSize: 26_214_400
```

Comment:
will also need hidden fields here for the files that are already saved, but i think leave that in this PR. we can get the form working with a create first, and then add the extra data and components needed for editing existing records in another PR